ch_pipeline/analysis/flagging.py

```diff
@@ -2260,3 +2261,195 @@ def process(self, data):
+        Default: 30.
+
+    Returns
+    -------
+    rainfall : np.ndarray
+        Cumulative rainfall totals, in mm.
+    """
+
+    # Extra buffer (in s) for reading rainfall measurements, to ensure that range of
+    # input times is always fully within range of rainfall timestamps
+    _TIME_BUFFER = 600
+
+    # Compute accumulation time in seconds
+    accumulation_time_s = accumulation_time * 3600
+
+    # Load rainfall measurements within relevant time range
+    rain_timestamps, rain_meas = load_rainfall(
+        times[0] - accumulation_time_s * 3600 - _TIME_BUFFER,
+        times[-1] + _TIME_BUFFER,
+        node_spoof,
+    )
+
+    # Compute number of rainfall timestamps to accumulate
+    dtimestamp = np.median(np.diff(rain_timestamps))
+    n_sum = np.rint(accumulation_time_s / dtimestamp).astype(int)
+
+    def _sum_previous_N(arr, N):
+        # For each element in an array, compute the sum of the element and the N-1
+        # previous elements
+
+        # For each element, compute the sum of the element and all previous elements
+        cumsum = np.cumsum(arr)
+
+        # Make new array to stor results
+        sums = np.zeros_like(arr)
+        # First N sums will just be equal to cumsum result
+        sums[:N] = cumsum[:N]
+        # Final sums will be difference of cumsum results separated by N entries
+        sums[N:] = cumsum[N:] - cumsum[: len(arr) - N]
+
+        return sums
+
+    # Compute cumulative rainfall totals at same timestamps
+    cumu_rainfall = _sum_previous_N(rain_meas, n_sum)
+
+    # For each input time, assign cumulative rainfall from first timestamp
+    # that occurs after this time. This errs on the side of potentially
+    # overestimating the cumulative rainfall at a given input time.
+    time_timestamp_idx = np.searchsorted(rain_timestamps, times, side="left")
+    cumu_rainfall = cumu_rainfall[time_timestamp_idx]
+
+    return cumu_rainfall
+
+
+class FlagRainfall(task.SingleTask):
+    """Flag times following periods of heavy rainfall.
+
+    This task uses rainfall measurements from the DRAO weather station to compute
+    the accumulated rainfall within a given time interval prior to each time sample.
+    If the rainfall total exceeds some threshold, the weight dataset is set to zero
+    for that time.
+
+    Parameters
+    ----------
+    accumulation_time : float
+        Number of hours over which to compute accumulated rainfall for each time sample.
+        Default: 30.
+    threshold : float
+        Rainfall threshold (in mm) for flagging. Default: 1.0.
+    node_spoof : dictionary
+        Host and directory for finding weather data.
+        Default: {'cedar_online': '/project/rpp-krs/chime/chime_online/'}
+    """
+
+    accumulation_time = config.Property(proptype=float, default=30.0)
+    threshold = config.Property(proptype=float, default=1.0)
+    node_spoof = config.Property(proptype=dict, default=_DEFAULT_NODE_SPOOF)
+
+    def process(self, timestream):
```

Comment:
Extremely nitpicky, but since this task can accept a sidereal stream as well as a time stream, using the variable name `timestream` could be confusing. Probably better to just use `datastream`, `data`, or something like that